py/map.c

```diff
@@ -126,6 +126,15 @@ STATIC void mp_map_rehash(mp_map_t *map) {
     m_del(mp_map_elem_t, old_table, old_alloc);
 }
 
+#if MICROPY_OPT_MAP_CACHING
+uint8_t map_lookup_cache[128];
+#define MAP_CACHE_OFFSET ((((uintptr_t) + (uintptr_t)index) >> 2) % sizeof(map_lookup_cache))
+#define MAP_CACHE_GET(map) (&(map)->table[map_lookup_cache[MAP_CACHE_OFFSET] % (map)->alloc])
+#define MAP_CACHE_SET(map, pos) map_lookup_cache[MAP_CACHE_OFFSET] = (pos) & 0xff;
```

Comment:
these macros should all take all arguments that they use, so there are no hidden arguments